tools/config-gen/main.go

```diff
@@ -40,26 +41,27 @@ type templateData struct {
 
 func validateFlags() error {
 	if *paramsFile == "" {
-		return fmt.Errorf("input filename cannot be empty")
+		return fmt.Errorf("params filename cannot be empty")
 	}
-	if *outFile == "" {
-		return fmt.Errorf("output filename cannot be empty")
+	if *outDir == "" {
+		return fmt.Errorf("output directory cannot be empty")
 	}
-	if *templateFile == "" {
-		return fmt.Errorf("template filename cannot be empty")
+	if *templateDir == "" {
+		return fmt.Errorf("template directory cannot be empty")
 	}
 	return nil
 }
 
-func write(data templateData) error {
-	outputFile, err := os.Create(*outFile)
+func write(data any, outF, tmplF string) error {
```

Comment:
Can you put more descriptive names to these parameters. 
here data is of any type, it will not be easy to understand what data is getting passed and why is it of any type?

Reply:
Changed the names to be more descriptive.